idk/kafka/cmd_test.go

```diff
@@ -682,3 +702,122 @@ func tDoHTTPPost(t *testing.T, url, contentType, body string) string {
+	consumer.Topics = []string{topic}
+	consumer.BatchSize = 1000
+	consumer.IDField = "user_id"
+	consumer.KafkaGroupInstanceId = "some_id"
+	consumer.ConsumerCloseTimeout = 3
+	consumer.KafkaMaxPollInterval = "15000"
+	consumer.MaxMsgs = 10000
+	consumer.Index = index
+	consumer.KafkaSessionTimeout = "10000"
+	consumer.LogPath = log_path
+
+	// run datagen and capture stdout and stderr in case of error
+	var datagen_0_stdout, datagen_0_stderr bytes.Buffer
+	cmdString := fmt.Sprintf("go run ../cmd/datagen/main.go --source %s --target %s --kafka.topic %s --kafka.subject %s --end-at %d --kafka.confluent-command.schema-registry-url %s --kafka.confluent-command.kafka-bootstrap-servers %s", source, target, topic, subject, genNumRecords, consumer.SchemaRegistryURL, consumer.KafkaBootstrapServers[0])
+	cmd := exec.Command("bash", "-c", cmdString)
+	cmd.Stdout = &datagen_0_stdout
+	cmd.Stderr = &datagen_0_stderr
+	err = cmd.Run()
+	if err != nil {
+		t.Errorf("stdout: %s", datagen_0_stdout.String())
+		t.Errorf("stderr: %s", datagen_0_stderr.String())
+		t.Fatalf("issue generating records for kafka: %s", err)
+	}
+
+	// run the consumer
+	consumerError := make(chan error)
+	go func() {
+		consumerErr := consumer.Run()
+		consumerError <- consumerErr
+	}()
+
+	// wait until all 5000 messages to be consumed
+	complete := false
+	start := time.Now()
+
+	for !complete {
+		buf, _ := os.ReadFile(log_path)
+		s := string(buf)
+		if strings.Contains(s, "records processed 0-> (5000)") {
+			complete = true
+		}
+		elapsed := time.Since(start)
+		if elapsed > 15*1000*1000*1000 { //15 seconds
+			t.Fatalf("unable to read from consumer log file")
+		}
+	}
+
+	// take a transaction lock on the database which drives
+	// max.poll.interval.ms timeout to occur
+	client := consumer.PilosaClient()
+	status, body, err := client.HTTPRequest("POST", "/transaction", []byte("{\"timeout\": 20, \"exclusive\": true, \"active\": true}"), nil)
+	if err != nil {
+		t.Fatalf("error taking transaction lock")
+	}
+	if status != 200 || !strings.Contains(string(body), "active\":true") {
+		t.Fatalf("was unable to successfully take transaction lock: %s", string(body))
+	}
+
+	// produce to kafka again capturing stdout and stderr in case of error
+	cmd = exec.Command("bash", "-c", cmdString)
+
+	var datagen_1_stdout, datagen_1_stderr bytes.Buffer
+
+	cmd.Stdout = &datagen_1_stdout
+	cmd.Stderr = &datagen_1_stderr
+	err = cmd.Run()
+	if err != nil {
+		t.Errorf("stdout: %s", datagen_1_stdout.String())
+		t.Errorf("stderr: %s", datagen_1_stderr.String())
+		t.Fatalf("issue generating records for kafka: %s", err)
+	}
+
+	// wait for the consumer to exit or 30 seconds to pass
+	select {
+	case <-consumerError:
+		buf, err := os.ReadFile(log_path)
+		if err != nil {
+			t.Errorf("issue reading consumer log: %s", err)
+		}
+		s := string(buf)
+		if !strings.Contains(s, "Application maximum poll interval") && !strings.Contains(s, "Unable to properly close consumer") {
```

Comment:
is the intent here that it must contain both, or that it must contain either? I believe this will only report a test-failure if both strings are missing, but I'd have thought we wanted to fail if either was missing.